metricflow/inference/rule/defaults.py

```diff
@@ -0,0 +1,78 @@
+import re
+from typing import List
+
+from metricflow.inference.rule.base import InferenceRule, InferenceSignalConfidence, InferenceSignalType
+from metricflow.inference.rule.rules import ColumnRegexMatcherRule
+
+
+class RuleDefaults:
+    """Static factory class for sensible default rules."""
+
+    # This is the default regex pattern that is used to determine if columns are identifiers.
+    # It simply matches column names ending with "id", case insensitive.
+    #
+    # We searched for words ending with "id" just to assess the chance of this resulting in a
+    # false positive. Our guess is most of those words would rarely, if ever, be used as column names.
+    # Therefore, not adding a mandatory "_" before "id" would benefit the product by matching names
+    # like "userid", despite the rare "squid", "mermaid" or "android" matches.
+    #
+    # See: https://www.thefreedictionary.com/words-that-end-in-id
+    ANY_IDENTIFIER_REGEX_PATTERN = re.compile(r"id$", flags=re.IGNORECASE)
```

Comment:
I know it is broad, but my rationale is that it's better to match stuff like `userid` and `customerid` at the expense of having a couple of (hopefully rare) false positives like `is_livid` and `grid` than to not match them and have the user type them in everytime. 

It will be very difficult to get a solution that gets everything right everytime, but IMO that's a tolerable compromise (given we make sure the user reviews everything to catch those misses).